api/factory/ginwebserver/common.go

```diff
@@ -0,0 +1,83 @@
+package ginwebserver
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/ElrondNetwork/elrond-go/api/logs"
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/gin-gonic/gin"
+	"github.com/gin-gonic/gin/binding"
+	"github.com/gorilla/websocket"
+	"gopkg.in/go-playground/validator.v8"
+)
+
+func checkArgs(args ArgsNewGinWebServerHandler) error {
+	baseError := errors.New("cannot create gin web server handler")
+	errHandler := func(details string) error {
+		return fmt.Errorf("%w: %s", baseError, details)
+	}
+
+	if check.IfNil(args.Facade) {
+		return errHandler("nil facade")
```

Comment:
for now, it seems like this. But if we would have a larger set of arguments, then it would be better to see `errHandler("error message")` n times instead of `fmt.Errorf("%w: error message", baseErrorMessage)`